lightning/backend/local.go

```diff
@@ -1193,50 +1196,35 @@ func (local *local) FetchRemoteTableModels(ctx context.Context, schemaName strin
 	return fetchRemoteTableModelsFromTLS(ctx, local.tls, schemaName)
 }
 
-func (local *local) WriteRows(
-	ctx context.Context,
-	engineUUID uuid.UUID,
-	tableName string,
-	columnNames []string,
-	ts uint64,
-	rows Rows,
-) (finalErr error) {
-	kvs := rows.(kvPairs)
-	if len(kvs) == 0 {
-		return nil
-	}
+func (local *local) MakeEmptyRows() Rows {
+	return kvPairs(nil)
+}
+
+func (local *local) NewEncoder(tbl table.Table, options *SessionOptions) (Encoder, error) {
+	return NewTableKVEncoder(tbl, options)
+}
 
+func (local *local) LocalWriter(ctx context.Context, engineUUID uuid.UUID) (EngineWriter, error) {
 	e, ok := local.engines.Load(engineUUID)
 	if !ok {
-		return errors.Errorf("could not find engine for %s", engineUUID.String())
+		return nil, errors.Errorf("could not find engine for %s", engineUUID.String())
 	}
 	engineFile := e.(*LocalFile)
-
-	// write to pebble to make them sorted
-	wb := engineFile.db.NewBatch()
-	defer wb.Close()
-	wo := &pebble.WriteOptions{Sync: false}
-
-	size := int64(0)
-	for _, pair := range kvs {
-		wb.Set(pair.Key, pair.Val, wo)
-		size += int64(len(pair.Key) + len(pair.Val))
-	}
-	if err := wb.Commit(wo); err != nil {
-		return err
+	tmpPath := filepath.Join(local.localStoreDir, engineUUID.String())
```

Comment:
This dir is already used in L#417 for open pebble db, I think better change to another dir. And btw, since all `LocalWriter`s share the same dir, just init it in the `OpenEngine` function is better

Reply:
OK